src/Diagrams/Swimunit/Dotmatrix.hs

```diff
@@ -0,0 +1,843 @@
+{-# LANGUAGE NoMonomorphismRestriction #-}
+
+module Diagrams.Swimunit.Dotmatrix
+where
+
+import qualified Data.Map as Map
+
+import Diagrams.Prelude
+import Diagrams.Backend.SVG.CmdLine
```

Comment:
Hi Chris,

wow. I agree that the interdependency of contrib or any other module to 
a specific backend should be kept at minimum.
But I fear that your suggestion is far above my understanding of 
Diagrams so that I probably will not be able to realize it without help.

A thought on dependency:
During first developing steps of the plotting library I realized that 
detecting errors is sometimes hard -- eg. not enough ticks, empty lists, 
... --
I dumped some text in the plots but since they take no space it was 
nearly impossible to scale them in the right way.
The Dotmatrix module solves this problem since the errors can be dumped 
as diagrams and are always scaled right
independent of any backend or third party font.

So it might be a good idea to separate plotting from Dotmatrix since it 
is a hack of its own.

Regards
     Wolfram


On 18.09.2015 20:16, Chris wrote:
>
> In src/Diagrams/Swimunit/Dotmatrix.hs 
> <https://github.com/diagrams/diagrams-contrib/pull/55#discussion_r39884749>:
>
> > @@ -0,0 +1,843 @@
> > +{-# LANGUAGE NoMonomorphismRestriction #-}
> > +
> > +module Diagrams.Swimunit.Dotmatrix
> > +where
> > +
> > +import qualified Data.Map as Map
> > +
> > +import Diagrams.Prelude
> > +import Diagrams.Backend.SVG.CmdLine
>
> |contrib| modules shouldn't depend on a specific backend. Instead of 
> |Diagram B| it should be something like |TypeableFloat n => QDiagram b 
> V2 n Any -> Dotfont -> String -> QDiagram b V2 n Any|.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/diagrams/diagrams-contrib/pull/55/files#r39884749>.
>